Components/Bites/src/OgreApplicationContextSDL.cpp

```diff
@@ -68,7 +72,36 @@ NativeWindowPair ApplicationContextSDL::createWindow(const Ogre::String& name, O
     p.miscParams["sdlwin"] = Ogre::StringConverter::toString(size_t(ret.native));
 
 #if OGRE_PLATFORM == OGRE_PLATFORM_LINUX
-    p.miscParams["externalWindowHandle"] = Ogre::StringConverter::toString(size_t(wmInfo.info.x11.window));
+    if (wmInfo.subsystem == SDL_SYSWM_WAYLAND)
+    {
+#ifdef OGRE_WAYLAND
+        Ogre::LogManager::getSingleton().logMessage("[SDL] Creating Wayland window");
+        wl_egl_window *win_impl = (wl_egl_window*)SDL_GetWindowData(ret.native, "wl_egl_window");
+
+        // https://github.com/libsdl-org/SDL/issues/5386
+        // https://github.com/bkaradzic/bgfx/blob/00fa5ad179f5aa13c1e44d0bcbccdc535aba2d00/examples/common/entry/entry_sdl.cpp#L53-L64
+        if(!win_impl)
+        {
+          Ogre::LogManager::getSingleton().logMessage("[SDL] Creating wl_egl_window");
+          int width, height;
+          SDL_GetWindowSize(ret.native, &width, &height);
+          wl_surface* surface = wmInfo.info.wl.surface;
+          win_impl = wl_egl_window_create(surface, width, height);
```

Comment:
I think we should do this unconditionally in `WaylandEGLWindow`. This way we dont need to use `SDL_GetWindowData` at all and we OgreBites will not depend on Wayland.

Reply:
Done.